programs/locker/src/instructions/v3/create_vesting_escrow.rs

```diff
@@ -0,0 +1,161 @@
+use anchor_spl::token::spl_token;
+use anchor_spl::token_2022::spl_token_2022;
+use anchor_spl::token_interface::{Mint, TokenAccount, TokenInterface};
+use vesting_escrow_v3::VestingEscrowV3;
+
+use crate::util::{
+    calculate_transfer_fee_included_amount, parse_remaining_accounts, transfer_to_escrow_v2,
+    validate_mint, AccountsType, ParsedRemainingAccounts,
+};
+use crate::TokenProgramFlag::{UseSplToken, UseToken2022};
+use crate::*;
+
+#[derive(AnchorSerialize, AnchorDeserialize)]
+/// Accounts for [locker::create_vesting_escrow_v3].
+pub struct CreateVestingEscrowV3Parameters {
+    pub total_deposit_amount: u64,
+    pub cancel_mode: u8,
+    pub root: [u8; 32],
+}
+
+impl CreateVestingEscrowV3Parameters {
+    pub fn init_escrow(
+        &self,
+        vesting_escrow: &AccountLoader<VestingEscrowV3>,
+        token_mint: Pubkey,
+        sender: Pubkey,
+        base: Pubkey,
+        total_deposit_amount: u64,
+        escrow_bump: u8,
+        token_program_flag: TokenProgramFlag,
+    ) -> Result<()> {
+        require!(
+            CancelMode::try_from(self.cancel_mode).is_ok(),
+            LockerError::InvalidCancelMode,
+        );
+
+        let mut vesting_escrow = vesting_escrow.load_init()?;
+        vesting_escrow.init(
+            token_mint,
+            sender,
+            base,
+            total_deposit_amount,
+            self.root,
+            self.cancel_mode,
+            escrow_bump,
+            token_program_flag.into(),
+        );
+
+        Ok(())
+    }
+}
+
+#[event_cpi]
+#[derive(Accounts)]
+pub struct CreateVestingEscrowV3<'info> {
+    #[account(mut)]
```

Comment:
base dont need to be mutable

Reply:
i duplicated it from v1 & v2
btw, i removed it.